x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx

```diff
@@ -0,0 +1,379 @@
+      // inject the reference
+      const dataViewIdRef = state.references?.find(
+        (ref) => ref.name === FIELD_STATS_DATA_VIEW_REF_NAME
+      );
+      // if the serializedState already contains a dataViewId, we don't want to overwrite it. (Unsaved state can cause this)
+      if (dataViewIdRef && serializedState && !serializedState.dataViewId) {
+        serializedState.dataViewId = dataViewIdRef?.id;
+      }
+      return serializedState;
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const [coreStart, pluginStart] = await getStartServices();
+
+      const { http, uiSettings, notifications, ...startServices } = coreStart;
+      const { lens, data, usageCollection, fieldFormats } = pluginStart;
+
+      const deps = {
+        http,
+        uiSettings,
+        data,
+        notifications,
+        lens,
+        usageCollection,
+        fieldFormats,
+        ...startServices,
+      };
+      const {
+        api: timeRangeApi,
+        comparators: timeRangeComparators,
+        serialize: serializeTimeRange,
+      } = initializeTimeRange(state);
+
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+
+      const {
+        fieldStatsControlsApi,
+        dataLoadingApi,
+        fieldStatsControlsComparators,
+        serializeFieldStatsChartState,
+        onFieldStatsTableDestroy,
+        resetData$,
+      } = initializeFieldStatsControls(state);
+      const { onError } = dataLoadingApi;
+
+      const defaultDataViewId = await deps.data.dataViews.getDefaultId();
+      const validDataViewId: string =
+        isDefined(state.dataViewId) && state.dataViewId !== ''
+          ? state.dataViewId
+          : defaultDataViewId ?? '';
+      let initialDataView: DataView[] | undefined;
+      try {
+        const dataView = await deps.data.dataViews.get(validDataViewId);
+        initialDataView = [dataView];
+      } catch (error) {
+        // Only need to publish blocking error if viewtype is data view, and no data view found
+        if (state.viewType === FieldStatsInitializerViewType.DATA_VIEW) {
+          onError(error);
+        }
+
+        if (isESQLQuery(state.query)) {
+          const indexPatternFromQuery = getIndexPatternFromESQLQuery(state.query.esql);
+
+          const adHocDataView = await getESQLAdHocDataview(
+            indexPatternFromQuery,
+            deps.data.dataViews
+          );
+          initialDataView = [adHocDataView];
+        }
+      }
+
+      const dataViews$ = new BehaviorSubject<DataView[] | undefined>(initialDataView);
+
+      const subscriptions = new Subscription();
+      if (fieldStatsControlsApi.dataViewId$) {
+        subscriptions.add(
+          fieldStatsControlsApi.dataViewId$
+            .pipe(
+              skip(1),
+              skipWhile((dataViewId) => !dataViewId && !defaultDataViewId),
+              switchMap((dataViewId) => deps.data.dataViews.get(dataViewId ?? defaultDataViewId))
```

Comment:
Looks like there is still a unhandled throw here.

Reply:
Updated here https://github.com/elastic/kibana/pull/184030/commits/0a39eb7364e235191c4f416dc3881afa8ef510e0